test/powershell/engine/DefaultCommands.Tests.ps1

```diff
@@ -528,4 +528,225 @@ Describe "Verify approved aliases list" -Tags "CI" {
+
+        .PARAMETER expression
+            The optional command expression to pass to sudo
+
+        .PARAMETER UsePowerShell
+            The expression should be use powershell
+
+        .EXAMPLE
+            InvokeSudo -expression 'Get-Item /etc' -usePowerShell
+
+        .NOTES
+        General notes
+        #>
+        #>
+        function InvokeSudo
+        {
+            param
+            (
+                [string] $SudoArgs = $null,
+                [string] $Expression = $null,
+                [switch] $UsePowerShell
+            )
+
+            $sb = [System.Text.StringBuilder]::new()
+            $null = $sb.Append('sudo')
+
+            if (-not [string]::IsNullOrEmpty($sudoArgs))
+            {
+                $null = $sb.Append(' ')
+                $null = $sb.Append($sudoArgs)
+            }
+
+            if (-not [string]::IsNullOrEmpty($Expression))
+            {
+                $null = $sb.Append(' ')
+                if ($UsePowerShell)
+                {
+                    # avoid expression evaluation
+                    $null = $sb.AppendLine('@"')
+                }
+                $null = $sb.Append($Expression)
+                if ($UsePowerShell)
+                {
+                    $null = $sb.AppendLine()
+                    $null = $sb.Append('"@')
+                }
+            }
+            $scriptBlock = [ScriptBlock]::Create($sb.ToString())
+
+            # build the command line arguments that the function should pass to sudo
+            $null = $sb.Clear()
+            if (-not [string]::IsNullOrEmpty($sudoArgs))
+            {
+                $null = $sb.Append($sudoArgs)
+            }
+
+            if (-not [string]::IsNullOrEmpty($Expression))
+            {
+                if ($sb.Length -gt 0)
+                {
+                    $null = $sb.Append(' ')
+                }
+                if ($UsePowerShell)
+                {
+                    $bytes = [System.Text.Encoding]::Unicode.GetBytes($Expression)
+                    $arguments = [Convert]::ToBase64String($bytes)
+                    $null = $sb.AppendFormat('powershell -encodedCommand {0}', $arguments)
+                }
+                else
+                {
+                    $null = $sb.Append($Expression)
+                }
+            }
+            # invoke the sudo function
+            $output = & $scriptBlock
+            # return the expected and actual arguments passed to sudo
+            return @{ExpectedArguments = $sb.ToString(); ActualArguments = $output}
+        }
+
+        It "Validates sudo parameters formatting using the mock" -Skip:$skipTest {
+            $result = InvokeSudo -SudoArgs '--verify'
+            $result.ActualArguments | Should Be $result.ExpectedArguments
+        }
+
+        It "Validates sudo parameters and expression formatting using the mock" -Skip:$skipTest {
+            $result = InvokeSudo -SudoArgs '--verify' -Expression 'ls /etc'

```

Comment:
I would expect the results to be the same; regardless.  I'm not actually executing the command; I'm verifying that the function detects that 'ls' is a native command and does not attempt to invoke PowerShell for it.
I'll add a comment to clarify.